src/libData/AccountData/AccountStoreBase.tpp

```diff
@@ -0,0 +1,379 @@
+                                      const Account& account)
+{
+    LOG_MARKER();
+
+    if (!DoesAccountExist(address))
+    {
+        m_addressToAccount->insert(make_pair(address, account));
+        // UpdateStateTrie(address, account);
+    }
+}
+
+template<class MAP>
+void AccountStoreBase<MAP>::AddAccount(const PubKey& pubKey,
+                                      const Account& account)
+{
+    AddAccount(Account::GetAddressFromPublicKey(pubKey), account);
+}
+
+template<class MAP>
+Account* AccountStoreBase<MAP>::GetAccount(const Address& address)
+{
+    auto it = m_addressToAccount->find(address);
+    if (it != m_addressToAccount->end())
+    {
+        return &it->second;
+    }
+    return nullptr;
+}
+
+template<class MAP> uint256_t AccountStoreBase<MAP>::GetNumOfAccounts() const
+{
+    LOG_MARKER();
+    return m_addressToAccount->size();
+}
+
+template<class MAP>
+bool AccountStoreBase<MAP>::IncreaseBalance(const Address& address,
+                                           const uint256_t& delta)
+{
+    // LOG_MARKER();
+
+    if (delta == 0)
+    {
+        return true;
+    }
+
+    Account* account = GetAccount(address);
+
+    LOG_GENERAL(INFO, "address: " << address);
+
+    if (account != nullptr && account->IncreaseBalance(delta))
+    {
+        // UpdateStateTrie(address, *account);
+        LOG_GENERAL(INFO, "account: " << *account);
+        return true;
+    }
+    // FIXME: remove this, temporary way to test transactions, should return false
+    else if (account == nullptr)
+    {
+        LOG_GENERAL(WARNING,
+            "AddAccount... FIXME: remove this, temporary way to test "
+            "transactions");
+        AddAccount(address, {delta, 0});
+        return true;
+    }
+
+    return false;
+}
+
+template<class MAP>
+bool AccountStoreBase<MAP>::DecreaseBalance(const Address& address,
+                                           const uint256_t& delta)
+{
+    // LOG_MARKER();
+
+    if (delta == 0)
+    {
+        return true;
+    }
+
+    Account* account = GetAccount(address);
+
+    LOG_GENERAL(INFO, "address: " << address);
+    LOG_GENERAL(INFO, "account: " << *account);
+
+    if (account != nullptr && account->DecreaseBalance(delta))
+    {
+        // UpdateStateTrie(address, *account);
+        return true;
+    }
+    // FIXME: remove this, temporary way to test transactions, should return false
+    else if (account == nullptr)
+    {
+        LOG_GENERAL(WARNING,
+                    "AddAccount... FIXME: remove this, temporary way to test "
+                    "transactions");
+        AddAccount(address, {10000000000, 0});
+        return true;
+    }
+
+    return false;
+}
+
+template<class MAP>
+bool AccountStoreBase<MAP>::TransferBalance(const Address& from,
+                                           const Address& to,
+                                           const uint256_t& delta)
+{
+    // LOG_MARKER();
+
+    if (DecreaseBalance(from, delta) && IncreaseBalance(to, delta))
```

Comment:
It's dangerous because some compiler would not execute right-hand side if left-hand side is false, or vice versa.
Please re-write this part to make sure the 2 function be executed.